### PR TITLE
Adds e2e test for compaction

### LINKED ISSUE #700: [Tests] Add compaction, copy-backups-task and secret-finalizer tests to e2e test suite
**Feature (What you would like to be added):**
Enhance e2e test suite to include compaction, copy-backups-task and secret-finalizer tests in addition to existing etcd and custodian tests.

**Motivation (Why is this needed?):**
Currently, druid e2e tests are used for testing the functionality of etcds, including snapshotting, restoration, scaling up and down the etcd cluster, zero-downtime upgrades, etc. It would be prudent to also include tests for compaction and EtcdCopyBackupsTasks, so that these need not be tested manually every time during release validation. 

**Approach/Hint to the implement solution (optional):**
- [x] #704
- [ ] #705
- [ ] #706

### LINKED ISSUE #704: Add compaction tests to e2e test suite
**Feature (What you would like to be added):**
Enhance e2e test suite to include compaction controller tests in addition to existing etcd controller tests.

**Motivation (Why is this needed?):**
Part of #700

### PR DESCRIPTION
**Pull request by abdasgupta**
**How to categorize this PR?**

/area control-plane
/kind test

**What this PR does / why we need it**:
This PR adds e2e test for compaction in druid

**Which issue(s) this PR fixes**:
Fixes #704 

**Special notes for your reviewer**:
As we don't have the new release of etcd backup-restore that supports on demand latest snapshot fetch from actual object storage, I used a custom image of etcd backup-restore that support on-demand latest snapshot from actual object storage. Please, delete the commit with the name `Uses a custom etcdbrctl image that supports latest snapshot pulled from actual storage.` from this PR, once a release is made for etcd backup-restore.

**Release note**:

```noteworthy developer
Added e2e test for compaction.
```

### CODE REVIEW

File: charts/images.yaml
Comment (seshachalam-yv) [inline charts/images.yaml:7]: Please revert it once BR is released

File: charts/images.yaml
Comment (seshachalam-yv) [inline charts/images.yaml:17]: Please revert it once BR is released

File: test/e2e/etcd_compaction_test.go
Comment (seshachalam-yv) [inline test/e2e/etcd_compaction_test.go:1]:
```suggestion
// Copyright (c) 2023 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file.
```

File: test/e2e/etcd_compaction_test.go
Comment (seshachalam-yv) [inline test/e2e/etcd_compaction_test.go:71]:
```suggestion
				It("should test compaction on backup", func() {
```

File: charts/druid/values.yaml
Comment (seshachalam-yv) [inline charts/druid/values.yaml:13]: Pls revert
```suggestion
# featureGates:
#   UseEtcdWrapper: true
```

File: skaffold.yaml
Comment (seshachalam-yv) [inline skaffold.yaml:28]: Can you remove the comments

File: test/e2e/etcd_compaction_test.go
Comment (seshachalam-yv) [inline test/e2e/etcd_compaction_test.go:105]: Instead of relying on `time.Sleep` for snapshot synchronization, a more robust approach would be to verify whether snapshots have been taken by checking the length of the snapshots. We can ascertain that new snapshots have been created if the count of snapshots post-operation is greater than the count observed prior to the operation. 

Otherwise, this can lead to flaky test 😢 
```go

Eventually(func() int {
    latestSnapshotsAfterOperations, err := getLatestSnapshots(kubeconfigPath, namespace, etcdName, debugPod.Name, debugPod.Spec.Containers[0].Name, 8080)
    Expect(err).NotTo(HaveOccurred())
    return len(latestSnapshotsAfterOperations.DeltaSnapshots)
}, timeoutDuration, pollingInterval).Should(BeNumerically(">", len(latestSnapshotsBeforePopulate.DeltaSnapshots)))

```

**Review by seshachalam-yv — CHANGES_REQUESTED** [review]
Thanks @abdasgupta for adding E2E tests

Please address my NIT comments

File: test/e2e/etcd_compaction_test.go
Comment (seshachalam-yv) [inline test/e2e/etcd_compaction_test.go:156]: Could you please rephrase the test message to reflect the expected outcome more accurately after a successful compaction? The current message should indicate that a full snapshot is expected to be taken post-compaction, and no delta snapshots should be present

File: test/e2e/etcd_compaction_test.go
Comment (abdasgupta) [reply test/e2e/etcd_compaction_test.go]: Done

File: test/e2e/etcd_compaction_test.go
Comment (abdasgupta) [reply test/e2e/etcd_compaction_test.go]: Done

File: charts/druid/values.yaml
Comment (abdasgupta) [reply charts/druid/values.yaml]: Done

File: test/e2e/etcd_compaction_test.go
Comment (abdasgupta) [reply test/e2e/etcd_compaction_test.go]: Done

**Review by seshachalam-yv — APPROVED** [review]
/lgtm

Merge this PR once etcd-backup-restore is released

**Comment by shreyas-s-rao** [comment]
@seshachalam-yv this PR will not pass until #726 is merged. Please rebase this on master once #726 has been merged

**Pull request merged**
